source/common/network/connection_impl.cc

```diff
@@ -179,6 +179,13 @@ void ConnectionImpl::noDelay(bool enable) {
   // Set NODELAY
   int new_value = enable;
   rc = setsockopt(fd_, IPPROTO_TCP, TCP_NODELAY, &new_value, sizeof(new_value));
+#ifdef __APPLE__
+  if (-1 == rc && errno == EINVAL) {
```

Comment:
Hm. This still happens and I think it's basically the same reason.

We call `ConnectionImpl::connect()` and then immediately `ConnectionImpl::noDelay(true)`, but the socket isn't connected yet, so OS X returns an error.

I added some code to read the value of TCP_NODELAY and it does get set despite the error.